contracts/Tokens/VAI/VAIController.sol

```diff
@@ -387,6 +394,38 @@ contract VAIController is VAIControllerStorageG2, VAIControllerErrorReporter, Ex
         return uint(Error.NO_ERROR);
     }
 
+    /**
+     * @notice Set the prime token contract address
+     * @param prime_ The new address of the prime token contract
+     * @return uint 0=success, otherwise a failure (see ErrorReporter.sol for details)
+     */
+    function _setPrimeToken(address prime_) external returns (uint) {
+        if (msg.sender != admin) {
+            return fail(Error.UNAUTHORIZED, FailureInfo.SET_COMPTROLLER_OWNER_CHECK);
+        }
+
+        emit NewPrime(prime, prime_);
+        prime = prime_;
+
+        return uint(Error.NO_ERROR);
+    }
+
+    /**
+     * @notice Toggle mint only for prime holder
+     * @param _mintEnabledOnlyForPrimeHolder set to true to allow minting only for prime holders
+     * @return uint 0=success, otherwise a failure (see ErrorReporter.sol for details)
+     */
+    function toggleOnlyPrimeHolderMint(bool _mintEnabledOnlyForPrimeHolder) external returns (uint) {
+        if (msg.sender != admin) {
+            return fail(Error.UNAUTHORIZED, FailureInfo.SET_COMPTROLLER_OWNER_CHECK);
+        }
```

Comment:
I would use `_ensureAllowed` instead, to be integrated with the ACM. In the VIP we should authorize Normal, Fast-track and Critical timelock to execute this function). And I wouldn't return any value in this function